force-app/main/default/classes/DMLOperations.cls

```diff
@@ -181,14 +224,16 @@ public with sharing class DMLOperations {
 	 */
 	public static void upsertAccountsWithContacts(List<Contact> contactsToUpsert) {
 		// Loop through each Contact
-
-		// Extract the Account Name from Contact's LastName
-
-		// Get the account based on the name or create one if it doesn't exist
-
-		// Link the Contact to its Account using the Account Id
-
+        for(Contact con : contactsToUpsert){
+            // Extract the Account Name from Contact's LastName
+            String accountName = con.LastName;
+            // Get the account based on the name or create one if it doesn't exist
+            Account acc = upsertAccount(accountName);
```

Comment:
This can cause DML inside of a for loop; how can you refactor using maps to avoid this scenario?